docs/src/test/java/docs/http/javadsl/server/directives/FutureDirectivesExamplesTest.java

```diff
@@ -138,12 +139,21 @@ public void testOnCompleteWithBreaker() throws InterruptedException {
           .assertStatusCode(StatusCodes.ServiceUnavailable())
           .assertEntity("The server is currently unavailable (because it is overloaded or down for maintenance).");
 
-    Thread.sleep(resetTimeout.toMillis() + 300);
-    // circuit breaker resets after this time
-    
-    testRoute(route).run(HttpRequest.GET("/divide/8/2"))
-      .assertEntity("The result was 4");
-    
+    Thread.sleep(resetTimeout.toMillis());
+
+    // circuit breaker resets after this time, but observing it
+    // is timing sensitive so retry a few times within a timeout
+    new TestKit(system()) {
+      {
+        awaitAssert(
+            FiniteDuration.create(300, TimeUnit.MILLISECONDS),
+            () -> {
+              testRoute(route).run(HttpRequest.GET("/divide/8/2"))
+                  .assertEntity("The result was 4");
+              return null;
+            });
+      }
+    };
```

Comment:
A bit messy in a code sample but I guess that's life in Java-land.

Reply:
Jep, unfortunate that testkit is not available and not easily mixed in Java.